src/api/che-api-client.ts

```diff
@@ -298,9 +296,33 @@ export class CheApiClient {
     }
   }
 
+  /**
+   * Returns Keycloak settings or undefined for single user mode.
+   */
+  async getKeycloakSettings(responseTimeoutMs = this.defaultCheResponseTimeoutMs): Promise<any | undefined> {
+    const endpoint = `${this.cheApiEndpoint}/keycloak/settings`
+    let response
+    try {
+      response = await this.axios.get(endpoint, { timeout: responseTimeoutMs })
+    } catch (error) {
+      if (error.response && error.response.status === 404) {
+        return
+      }
+      throw this.getCheApiError(error, endpoint)
+    }
+    if (!response || response.status !== 200 || !response.data) {
+      throw new Error('E_BAD_RESP_CHE_API')
```

Comment:
Maybe more informative? Include response code and message?

Reply:
ok, I'll add more information to the error. However, this check is more formal, I've never seen the error.